src/condor_scripts/condor_upgrade_check

```diff
@@ -559,274 +155,554 @@ def warn_job_router_syntax_deprecation():
+            continue
+
+        if len(only_list) > 0:
+            check_tag = True
+            if "WARNINGS" in only_list and check.is_warning:
+                check_tag = False
+            if check_tag and check.tag.upper() not in only_list:
+                continue
+
+        if args.ce and check.system == System.STANDARD:
+            continue
+        elif not args.ce and check.system == System.CE:
+            continue
+
+        if IS_WINDOWS and check.needs_root:
+            continue
+
+        if check.needs_root and os.geteuid() != 0:
+            format_print(
+                f"Warning: Skipping check '{check.name}' because it requires root.",
+            )
+            continue
+
+        if not args.all and check.version <= CURRENT_VERSION:
+            continue
+
+        if not args.warnings and check.is_warning:
+            continue
+
+        missing_daemon = False
+        for daemon in check.daemons:
+            if daemon.upper() not in daemons:
+                missing_daemon = True
+                break
+        if missing_daemon:
+            continue
+
+        num_checks_ran += 1
+
+        format_print(
+            f"Checking {check.name} [{check.tag}]:", newline=True, debug=Debug.DETAILED
+        )
+
+        if not check.test(args):
+            num_failed_checks += 1
+        else:
+            format_print("No issue found with check", offset=8, debug=Debug.DETAILED)
+
+    return (num_failed_checks, num_checks_ran)
+
+
+class NegateAction(argparse.Action):
+    def __call__(self, parser, args, values, option_string=None):
+        setattr(args, self.dest, option_string.rstrip("-").startswith("no"))
+
+
+def parse_args():
+    """Function to parse tool command line arguments"""
+    parser = argparse.ArgumentParser(
+        prog="condor_upgrade_check",
+        description=textwrap.dedent(
+            """
+            Tool to help check for known incompatabilites with a current
+            HTCondor install when upgrading to a new major version.
+
+            This should be ran on a host with HTCondor installed prior
+            to upgrading to new major version (V23 -> V24) to check current
+            installation for potential issues that may occur during an
+            upgrade.
+
+            Examples:
+                condor_upgrade_check
+                condor_upgrade_check --tags
+                condor_upgrade_check --dump
+                condor_upgrade_check -i BAR -vv
+                condor_upgrade_check -ce --only FOO BAR BAZ --no-warnings
+            """
+        ),
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-ce",
+        "--ce",
+        "-CE",
+        "--CE",
+        action="store_true",
+        help="Do checks for HTCondor-CE environment.",
+    )
+
+    parser.add_argument(
+        "-a",
+        "--all",
+        action="store_true",
+        help="Run all available checks (Ignores version check).",
```

Comment:
This is a little awkward, maybe...
```suggestion
        help="Run all available checks for all versions.",
```
I don't like that all that much better though.